pwnlib/tubes/ssh.py

```diff
@@ -441,14 +441,34 @@ def __init__(self, parent, host, port, *a, **kw):
         # keep the parent from being garbage collected in some cases
         self.parent = parent
 
+        # keep reference to tunnel process to avoid garbage collection
+        self.tunnel = None
+
         self.host  = parent.host
         self.rhost = host
         self.rport = port
 
+        import paramiko.ssh_exception
         msg = 'Connecting to %s:%d via SSH to %s' % (self.rhost, self.rport, self.host)
         with self.waitfor(msg) as h:
             try:
                 self.sock = parent.transport.open_channel('direct-tcpip', (host, port), ('127.0.0.1', 0))
+            except paramiko.ssh_exception.ChannelException as e:
+                # Workaround AllowTcpForwarding no in sshd_config
+                if e.args != (1, 'Administratively prohibited'):
+                    self.exception(str(e))
+                    raise e
+                
+                self.debug('Failed to open channel, trying to connect to remote port manually using netcat.')
+                if parent.which('nc'):
+                    ncat = 'nc'
+                elif parent.which('ncat'):
+                    ncat = 'ncat'
+                else:
+                    self.exception('Could not find ncat or nc on remote. Cannot connect to remote port.')
+                    raise
```

Comment:
```suggestion
                if parent.which('nc'):
                    ncat = 'nc'
                elif parent.which('ncat'):
                    ncat = 'ncat'
                elif parent.which('netcat'):
                    ncat = 'netcat'
                else:
                    self.exception('Could not find ncat, nc or netcat on remote. Cannot connect to remote port.')
                    raise
```
Some Linux distro calls it `netcat`. Although I can't remember which does, adding another fallback should be fine.

Reply:
Interesting, I never stumbled on a distro with just a "netcat" command. Would be nice to know which one names the command that way.

Reply:
> Interesting, I never stumbled on a distro with just a "netcat" command. Would be nice to know which one names the command that way.

Comments from https://unix.stackexchange.com/questions/368155:

> On RHEL/CentOS 7, there are /usr/bin/ncat (provided by the nmap-ncat RPM package) and /usr/bin/netcat (which is OpenBSD nc, provided by the netcat RPM package). And /usr/bin/nc is just a symbolic link to one of the two previous ones that can be managed on RHEL/CentOS 7 by alternatives --config nmap, and on RHEL 8 (or later) or derivatives by alternatives --config nc).

> Looking at the packets in Debian, there are two different implementations of netcat (nc), I think both of them install with netcat as well as nc symlinked to the same binary (Though they use Debian's "alternatives" system to control which one of the implementation appears with the main name.) Those two are [netcat-traditional](https://packages.debian.org/jessie/netcat-traditional) ("the "classic" netcat, written by Hobbit. It lacks many features found in netcat-openbsd.") and [netcat-openbsd](https://packages.debian.org/jessie/netcat-openbsd) ("the OpenBSD rewrite of netcat, including support for IPv6, proxies, and Unix sockets.").

Arch LInux also provides `/usr/bin/netcat`, which is symlinked to `nc`.